src/linker/Linker/Driver.cs

```diff
@@ -674,6 +674,13 @@ public bool Run (ILogger customLogger = null)
 
 			try {
 				p.Process (context);
+			} catch (Exception ex) {
+				Console.Error.WriteLine ("The illinker exited with error:");
+				if (ex is LinkerErrorException le) {
+					context.LogMessage (le.MessageContainer);
+				} else {
+					Console.Error.WriteLine (ex);
+				}
 			} finally {
```

Comment:
I think we need to return false here to produce a failing exit code (which signals failure to the MSBuild task, among other things).